File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/event/LayoutData.java
package io.github.orioncraftmc.meditate.internal.event;

import java.util.ArrayList;
import org.jetbrains.annotations.NotNull;

public class LayoutData extends CallableEvent //Type originates from: event.h
{
    public int layouts;
    public int measures;
    public int maxMeasureCache;
    public int cachedLayouts;
    public int cachedMeasures;
    public int measureCallbacks;
    public final @NotNull ArrayList<Integer> measureCallbackReasonsCount = new ArrayList<>();

    public LayoutData() {
        for (int i = 0; i < LayoutPassReason.COUNT.getValue(); i++) {
            measureCallbackReasonsCount.add(0);
        }
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/event/LayoutData.kt
package app.cash.redwood.yoga.event

class LayoutData : CallableEvent() //Type originates from: event.h
{
    var layouts = 0
    var measures = 0
    var maxMeasureCache = 0
    var cachedLayouts = 0
    var cachedMeasures = 0
    var measureCallbacks = 0
    val measureCallbackReasonsCount = mutableListOf<Int?>()

    init {
        for (i in 0 until LayoutPassReason.COUNT.getValue()) {
            measureCallbackReasonsCount.add(0)
        }
    }
}
